src/OpenSpaceToolkit/Mathematics/Objects/Interval.tpp

```diff
@@ -233,17 +233,55 @@ T                               Interval<T>::getUpperBound                  ( )
 
 }
 
-//                                 template <class T>
-// Interval<T>                     Interval<T>::getIntersectionWith            (   const   Interval&                   anInterval                                  ) const
-// {
-//     return Interval<T>::Undefined() ; // [TBI]
-// }
+                                template <class T>
+Interval<T>                     Interval<T>::getIntersectionWith            (   const   Interval&                   anInterval                                  ) const
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Interval") ;
+    }
 
-//                                 template <class T>
-// Interval<T>                     Interval<T>::getUnionWith                   (   const   Interval&                   anInterval                                  ) const
-// {
-//     return Interval<T>::Undefined() ; // [TBI]
-// }
+    if (!anInterval.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Interval") ;
+    }
+
+    if (anInterval.getType() != type_)
+    {
+        throw ostk::core::error::runtime::Wrong("Different types of interval") ;
+    }
+
```

Comment:
Looks good! Thanks @vishwaLoft for adding this! If you are leveraging this somewhere else and want to merge asap, LGTM. If we have a bit more time, we could probably adapt to resolve for all types combination, what do you think? Happy to take a pass at it!

Reply:
Thanks for the feedback man! This is not urgent. I think the way it stands... if we try and do an intersection of two different intervals, it's ambiguous which interval type to keep for the resulting interval...
Right?

Reply:
Hum, I was thinking sth along these lines for instance:
- A = [a, b[
- B = ]c, d[ with c in [a, b] and d > b
- then we'd get the intersection C = ]c, b[ for instance
and we can adapt for all the combinations nah? Not sure what you mean by ambiguous. The potential problematic case I am thinking of is when extremities are close (like d ~ b) to one another with less than machine precision level but in the case we can set a `tolerance` value nah?
Let me know if I am missing sth, I feel like these are well defined for unions and intersections 🤔

Reply:
Yeah I was referring to cases when d == b, which one do we keep closed or open. I guess we can always pick the most restrictive case.

Reply:
Oh man there's going to be like a bajillion combinations here X_X XD

Reply:
Okay yeah we'll have to do 6 if statements for each of the following:
    // closed <-> open
    // open <-> closed
    // halfOpenLeft <-> open
    // open <-> halfOpenLeft
    // halfOpenLeft <-> closed
    // closed <-> halfOpenLeft
    // halfOpenRight <-> open
    // open <-> halfOpenRight
    // halfOpenRight <-> closed
    // closed <-> halfOpenRight
    // halfOpenRight <-> halfOpenLeft
    // halfOpenLeft <-> halfOpenRight

How about for now we just let it be like this? And if the need arises for different intervals then we can add it?

Reply:
I can change the error to `Different Interval Types not yet supported`

Reply:
> How about for now we just let it be like this? And if the need arises for different intervals then we can add it?
The implementation is indeed a bit of a PITA. We can keep it like this if inter-types operations is not a priority for now, fine with me!